misc/test_runner/test_runner.c

```diff
@@ -118,12 +124,14 @@ int main() {
       perror("waitpid");
       return 2;
     }
-    if WIFSIGNALED (stat) {
+    if (WIFSIGNALED(stat) && WTERMSIG(stat) != test_info->signal) {
       fprintf(stderr, "forked test child was terminated by signal %d\n", WTERMSIG(stat));
       return 1;
     }
     int exit_status = WEXITSTATUS(stat);
-    if (exit_status != test_info->exit_code) {
+    if (exit_status != test_info->exit_code ||
+        // Sometimes the signal doesn't show up with `WIFSIGNALED`, so we check the exit status as well.
+        (exit_status > 128 && exit_status - 128 == test_info->signal)) {
```

Comment:
This seems wrong. The first half checks if the exit status is not what's expected (which is a problem). We or this against the exit status indicating we received the expected signal (which is not a problem).